prompt/prompt-tokenizer/Module.md

```diff
@@ -0,0 +1,54 @@
+# Module prompt-tokenizer
+
+A module that provides interfaces and implementations for tokenizing text and counting tokens when working with Large Language Models (LLMs).
+
+### Overview
+
+The prompt-tokenizer module offers tools for estimating the number of tokens in text strings, which is essential for working with LLMs that have token limits or require token counting. The module includes:
+
+- A `Tokenizer` interface that defines the contract for token counting
+- Multiple implementations with varying levels of accuracy and performance:
+  - `NoTokenizer`: A minimal implementation that always returns 0 (useful when token counting is not needed)
+  - `RegexBasedTokenizer`: A simple implementation that provides reasonable approximations for most LLMs
+  - `TiktokenEncoder`: An advanced implementation based on OpenAI's tiktoken library, using byte pair encoding (BPE) for more accurate token counting
+
+This module is particularly useful for LLMs like Ollama that don't provide token counts in their responses, requiring client-side estimation.
```

Comment:
This is not true for Ollama though, as it turned out, right? Ollama should provide token information, as was mentioned in the conversation above? (link: https://github.com/ollama/ollama/blob/main/docs/api.md#response)